src/Articulate/PathHelper.cs

```diff
@@ -38,5 +39,15 @@ public static string GetThemePartialViewPath(IMasterModel model, string viewName
             var path = "~/App_Plugins/Articulate/Themes/{0}/Views/Partials/{1}.cshtml";
             return string.Format(path, model.Theme, viewName);
         }
+
+        /// <summary>
+        /// Get the full domain of the current page
+        /// </summary>
+        public static string GetDomain()
+        {
+            return HttpContext.Current.Request.Url.Scheme +
+                System.Uri.SchemeDelimiter +
+                HttpContext.Current.Request.Url.Authority;
```

Comment:
Again, any use of singletons will not be accepted, the Request object should be either passed in as a parameter